specification/marketplace/resource-manager/Microsoft.Marketplace/stable/2021-12-01/Marketplace.json

```diff
@@ -0,0 +1,2894 @@
+          "description": "Gets request date",
+          "format": "date-time",
+          "readOnly": true
+        },
+        "justification": {
+          "description": "Gets or sets user's justification for the plan's request",
+          "type": "string"
+        },
+        "subscriptionId": {
+          "description": "Gets or sets the subscription id that the user is requesting to add the plan to",
+          "type": "string"
+        },
+        "subscriptionName": {
+          "description": "Gets or sets the subscription name that the user is requesting to add the plan to",
+          "type": "string"
+        }
+      }
+    },
+    "QueryRequestApproval": {
+      "description": "Gets the request plans with indication on each plan whether is approved by the admin, has pending request or not requested yet",
+      "type": "object",
+      "properties": {
+        "uniqueOfferId": {
+          "description": "Gets or sets unique offer id.",
+          "type": "string"
+        },
+        "plansDetails": {
+          "description": "Gets or sets the plans details",
+          "type": "object",
+          "additionalProperties": {
+            "$ref": "#/definitions/planDetails"
+          }
+        },
+        "etag": {
+          "description": "Gets or sets e-tag field ",
+          "type": "string"
+        },
+        "messageCode": {
+          "description": "Gets or sets the notification message id",
+          "type": "integer",
+          "format": "int64"
+        }
+      }
+    },
+    "QueryRequestApprovalProperties": {
+      "description": "The details to get the request plans statuses",
+      "type": "object",
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/RequestDetails",
+          "description": "The details to get the request plans statuses"
+        }
+      }
+    },
+    "RequestDetails": {
+      "description": "Request details needed to get the plans statuses",
+      "type": "object",
+      "properties": {
+        "publisherId": {
+          "description": "The offer's publisher id",
+          "type": "string"
+        },
+        "planIds": {
+          "description": "Current plans list",
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "subscriptionId": {
+          "description": "Gets or sets the subscription id",
+          "type": "string"
+        }
+      }
+    },
+    "NewPlansNotificationsList": {
+      "description": "List of all new plans notifications for public offers",
+      "type": "object",
+      "properties": {
+        "newPlansNotifications": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/NewNotifications"
+          }
+        }
+      }
+    },
+    "SubscriptionsContextList": {
+      "description": "Get private store subscriptions context",
+      "type": "object",
+      "properties": {
+        "subscriptionsContext": {
```

Comment:
>subscriptionsContext [](http://example.com/codeflow?start=9&length=20)

Is there a better name/description for this? Not clear what a subscriptionsContext is. Also, it needs to be plural (subscriptionsContexts?) since it's a list. From the related descriptions, it sounds like this is just a list of subscriptions, so why not "subscriptionIds" (presuming these are subscription Ids)?

Reply:
fixed